devconf-ironic-and-edgy/source/index.rst

```diff
@@ -0,0 +1,380 @@
+
+* Converged infrastructure management e.g. servers, switches, storage, power
+* Highly capable BMCs e.g. BIOS, OOB monitoring
+* Reliable and secure management protocols
+
+.. Things to talk about ^
+
+  Not specifically driven by the edge effort, rather for simplification
+  and cutting costs, hardware management tech tends to converge into
+  using common protocols and models. Now days Redfish (incorporting
+  NETCONF) serves as such a common ground for everything hardware
+  e.g. computers, switches and storage devices.
+
+  The BMCs, those small satellite computers that are always up and
+  running providing out-of-band access to the system being managed,
+  have evolved from a mere tiny controller to a powerful computer
+  capable to run heavy software.
+
+  The exposure of the inner system details has also grown a lot. The
+  modern BMCs can manage system BIOS, report system health and in
+  hardware configuration in great details.
+
+  The introduction of the Redfish hardware management protocol
+  greately improved the reliability and security of remote access
+  to the BMC and therefore to the hardware fleet.
+
+OpenStack Ironic
+================
+
+* Official OpenStack bare metal hypervisor since the *Kilo* cycle
+* Lively upstream community
+* Established relationships with hardware vendors
+
+.. Things to talk about ^
+
+  Ironic is the OpenStack project that implements a nova-managable
+  hypervisor targeting bare metal servers. The goal here is to
+  to treat bare metal machines as VMs from the user perspective.
+
+  Ironic has been concieved as a fork of nova baremetal driver since
+  OpenStack *Icehouse* cycle, by the *Kilo* cycle ironic has become
+  the officially integrated OpenStack project.
+
+  Ironic is already a relatively large project with quite active and
+  divierse community of users and contributors.
+
+  Targeting hardware management, ironic has managed to attract a
+  handful of high-profile hardware vendors thus creating and maintaining
+  vendor-specific *drivers* (AKA *hardware types*) interfacing ironic
+  with specific family of computers.
+
+Ironic at the Edge
+==================
+
+...
+
+Bare metal at the Edge
+======================
+
+Challenges:
+
+* Unreliable connection
+* Insecure networks
+* Low bandwidth
+
+.. Things to talk about ^
+
+Just to re-iterate similar slide from the beginning to set up the context
+for the next series of slides.
+
+Redfish to the rescue
+=====================
+
+Features:
+
+* Converged, aggregated management
+* Remote access over TCP
+* Ubiquitous TLS
+* Virtual Media boot
+* Composable hardware
+
+.. Things to talk about ^
+
+  Redfish is trying to solve many shortcomings that exist in the hardware
+  managenemt sphere. Luckily, many Redfish features play well in the
+  edge context.
+
+  In the following slides we are going to look into the relevant
+  Redfish feature and how they are being leveraged to solve the
+  edge use-case.
+
+Redfish: aggregated management
```

Comment:
Potentially related topic: conductor groups (locating conductors closer to WAN links to nodes).